src/components/dialog/dialog.component.js

```diff
@@ -361,32 +223,49 @@ class Dialog extends Modal {
       dialogProps['aria-describedby'] = 'carbon-dialog-subtitle';
     }
 
-    if (this.props.height) {
-      style.minHeight = `calc(${height} - 40px)`;
-    }
-
     return (
-      <div
+      <DialogStyle
         ref={ (d) => { this._dialog = d; } }
         { ...dialogProps }
         { ...this.componentTags(this.props) }
         onBlur={ this.onDialogBlur }
       >
         { this.dialogTitle }
 
-        <div className='carbon-dialog__content' ref={ (c) => { this._content = c; } }>
-          <div
-            className='carbon-dialog__inner-content' ref={ (c) => { this._innerContent = c; } }
-            style={ style }
+        <DialogContentStyle ref={ (c) => { this._content = c; } }>
+          <DialogInnerContentStyle
+            fixedBottom={ this.appliedFixedBottom }
+            ref={ (c) => { this._innerContent = c; } }
+            height={ this.props.height }
           >
             { this.props.children }
             { this.additionalContent() }
-          </div>
-        </div>
+          </DialogInnerContentStyle>
+        </DialogContentStyle>
         { this.closeIcon }
-      </div>
+      </DialogStyle>
     );
   }
 }
 
+Dialog.propTypes = assign({}, Modal.propTypes, {
```

Comment:
is there a reason to use `assign` instead of the spread operator?

Reply:
fixed